### DISCUSSION TITLE
Compatibility with standard PM / Custom Node Types

### DISCUSSION THREAD

**Discussion by RaphaelBuffl** [remirror/remirror#203] [General]
Hey there, 

thank you very much for this library! First of all I have to appologize, if the following question is stupid. We're currently migrating away from Slate due to the lack of Android support and API stability. This looks really promising. However we have two questions remaining: 

- Is remirror's Data format standard ProseMirror? That is, can we potentially interchange with standalone Prosemirror in a non react context? 
- Is it possible to use PM plugins, such as the table plugin? 
- Is it possible to build custom node types and render those as React Components. 

Thank a lot in advance, 
Raphael

**Comment by ifiokjr** [comment]
> Is remirror's Data format standard ProseMirror? That is, can we potentially interchange with standalone Prosemirror in a non react context?

Yes

> Is it possible to use PM plugins, such as the table plugin?

Yes and it's something that can be prioritised if requested by others. It's definitely something I'm looking at due to interest https://github.com/ifiokjr/remirror/issues/195

> Is it possible to build custom node types and render those as React Components.

**Yes** although I've avoided nodes where possible due to some inherent difficulties. The [emoji extension](https://github.com/ifiokjr/remirror/blob/canary/%40remirror/extension-emoji/readme.md) was initially a node extension but it suffered a lot on mobile and android.

**Comment by onoufriosm** [comment]
@ifiokjr 

> Is it possible to build custom node types and render those as React Components.

From my understanding this is a requirement to render a node as any react component (similar to slate), correct? (which would be extremely convenient, especially for everyone migrating from slate)

Is this something that is going to get better support in the **next** version?

**Comment by ifiokjr** [comment]
> Is this something that is going to get better support in the next version?


It's definitely possible, and the functionality will be retained. 

Personally, I've found fewer bugs when using the DOM to interact with anything inside `contenteditable` areas. Mobile support is easier to guarantee when working this way.
